src/app/modules/contributors/views/top-contributors/sources/sources.component.ts

```diff
@@ -1,27 +1,19 @@
 import { Component } from '@angular/core';
-import {AuthService, ErrorDetails, ErrorsManagerService, Report, ReportConfig, ReportService} from 'shared/services';
+import { AuthService, ErrorDetails, ErrorsManagerService, GraphsData, Report, ReportConfig, ReportService } from 'shared/services';
 import { map, switchMap } from 'rxjs/operators';
 import { of as ObservableOf } from 'rxjs';
 import { AreaBlockerMessage, AreaBlockerMessageButton } from '@kaltura-ng/kaltura-ui';
-import {
-  KalturaEndUserReportInputFilter,
-  KalturaFilterPager,
-  KalturaObjectBaseFactory, KalturaReportGraph,
-  KalturaReportInputFilter,
-  KalturaReportInterval,
-  KalturaReportTable, KalturaReportTotal,
-  KalturaReportType
-} from 'kaltura-ngx-client';
-import { ReportDataConfig } from 'shared/services/storage-data-base.config';
+import { KalturaEndUserReportInputFilter, KalturaFilterPager, KalturaObjectBaseFactory, KalturaReportGraph, KalturaReportInterval, KalturaReportTable, KalturaReportType } from 'kaltura-ngx-client';
+import { ReportDataConfig, ReportDataItemConfig } from 'shared/services/storage-data-base.config';
 import { TranslateService } from '@ngx-translate/core';
 import { CompareService } from 'shared/services/compare.service';
 import { SourcesDataConfig } from './sources-data.config';
 import { TrendService } from 'shared/services/trend.service';
 import { TopContributorsBaseReportComponent } from '../top-contributors-base-report/top-contributors-base-report.component';
 import { Tab } from 'shared/components/report-tabs/report-tabs.component';
-import * as moment from "moment";
-import {DateFilterUtils} from "shared/components/date-filter/date-filter-utils";
-import {isEmptyObject} from "shared/utils/is-empty-object";
+import { DateFilterUtils } from 'shared/components/date-filter/date-filter-utils';
```

Comment:
@diamond-darrell  - these imports are not used, please clear all unused imports

Reply:
OK

> @diamond-darrell - regarding the missing labels in the graph, indeed happens due to lack of space. I think that if we have more than 3 sources, we can set xAxis.axisLabel.rotate to 45. Let's give it a try and see how it looks.

fixed